brave/brave-core
# Add NTP widget, search suggestion setting and theme metrics

Resolves https://github.com/brave/brave-browser/issues/40569



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

For `Brave.Theme.BrowserColorScheme`:

0. Start with fresh profile. Ensure metric exists in local state with an answer that matches the browser setting.
1. Change the color mode setting in the appearance settings. Ensure the metric updates with the correct value.

For `Brave.Theme.ThemeColorDefault`:
(note that this metric reports on the last loaded profile, and NOT the last open profile)

0. Start with fresh profile. Ensure metric exists with value 1.
1. Create a new profile and select a different color scheme in the profile creation dialog.
2. Ensure metric has value 0.
3. Restart browser and open the original profile. Ensure metric is 1.

For `Brave.Search.SearchSuggest`:

0. Start with fresh profile. Ensure metric exists with value 0.
1. Enable the search suggestions setting in the search settings.
2. Ensure the metric has value 1.

For `Brave.Search.GoogleWidgetUsage`:

0. Start with fresh profile. Ensure metric does not exist.
1. Make a Brave search via the NTP search widget, ensure metric still does not exist.
2. Make a Google search via the NTP search widget, ensure metric exists with value 1.
